sdk/servicebus/azure-servicebus/azure/servicebus/aio/_base_handler_async.py

```diff
@@ -69,6 +93,17 @@ def __init__(
         self._auth_uri = None
         self._properties = create_properties(self._config.user_agent)
 
+    @classmethod
+    def _convert_connection_string_to_kwargs(cls, conn_str, **kwargs):
+        # pylint:disable=protected-access
+        return BaseHandlerSync._convert_connection_string_to_kwargs(conn_str, **kwargs)
+
+    @classmethod
+    def _create_credential_from_connection_string_parameters(cls, token, token_expiry, policy, key):
+        if token and token_expiry:
+            return ServiceBusSASTokenCredential(token, token_expiry)
+        return ServiceBusSharedKeyCredential(policy, key)
```

Comment:
not blocking: same suggestion as the above one, if we move this method into utils, then it could be shared between sync and async

Reply:
It was initially in utils, I put it here to abuse classmethods as a way to get sync/async dispatch; the other approach I saw offhand was to pass it as a functional callback (and have a sync vs async credential builder) but that seemed less intuitive.

Let me know if you see a clever way to sort this out that I'm being blind to. (the fact that credentials are distinct from sync vs async)

I did distinctly want to merge some of the flows between the mgmt and non-mgmt code (dataplane) for parsing conn str, so there's definitely some usability improvements here, but this particular one seemed like a bit of a rock/hardplace.

Reply:
I personally prefer the utils approach combined with functional callback (sync vs async credential builder).
But I need to write some code first.
anyways, it's just a matter of implementation detail, we can discuss later on the structure.

Reply:
https://github.com/Azure/azure-sdk-for-python/issues/13683